### PR TITLE
Icf create epi v2 client

### PR DESCRIPTION
**Pull request by DookTibs**
Add client operations to CRUD epiv2, and required tests.

### CODE REVIEW

File: client/hawc_client/epiv2.py [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+
+from .client import BaseClient
+
+
+class EpiV2Client(BaseClient):
+    """
+    Client class for epidemiology v2 requests.
+    """
+
+    """
```
Comment (shapiromatron) [inline client/hawc_client/epiv2.py:12]: uncomment these methods for metadata and data?

File: tests/client/test_client.py [hunk 2]
```diff
@@ -531,6 +533,304 @@ def test_epi_create(self):
...
+                adjustment_factor_id,
+                epi_client.delete_adjustment_factor,
+                epiv2models.AdjustmentFactor,
+            ),
+            (chemical_id, epi_client.delete_chemical, epiv2models.Chemical),
+            (exposure_id, epi_client.delete_exposure, epiv2models.Exposure),
+            (design_id, epi_client.delete_design, epiv2models.Design),
+        )
+
+        for deletion in deletions:
```
Comment (shapiromatron) [inline tests/client/test_client.py:826]: use tuple unpacking in for loop https://www.pythonmorsels.com/tuple-unpacking/

File: tests/client/test_client.py [hunk 3]
```diff
@@ -531,6 +533,304 @@ def test_epi_create(self):
...
+        data_extraction_result = epi_client.update_data_extraction(
+            data_extraction_id, data_extraction_payload
+        )
+        common_result_check(data_extraction_result, data_extraction_payload, "comments")
+
+        # read
+        data_extraction_result = epi_client.get_data_extraction(data_extraction_id)
+        common_result_check(data_extraction_result, data_extraction_payload, "comments")
+
+        ###
```
Comment (shapiromatron) [inline tests/client/test_client.py:803]: remove this comment

File: tests/client/test_client.py [hunk 4]
```diff
@@ -531,6 +533,304 @@ def test_epi_create(self):
...
+        # [0] a primary key
+        # [1] the client method to use to delete it
+        # [2] the model class
+
+        deletions = (
+            (data_extraction_id, epi_client.delete_data_extraction, epiv2models.DataExtraction),
+            (exposure_level_id, epi_client.delete_exposure_level, epiv2models.ExposureLevel),
+            (outcome_id, epi_client.delete_outcome, epiv2models.Outcome),
+            (
+                adjustment_factor_id,
```
Comment (shapiromatron) [inline tests/client/test_client.py:817]: shorten names so ruff formats to a single line

File: tests/client/test_client.py [hunk 5]
```diff
@@ -531,6 +533,304 @@ def test_epi_create(self):
         Exposure.objects.filter(id=exposure_id).delete()
         StudyPopulation.objects.filter(id=study_pop_id).delete()
 
+    #####################
+    # EpiV2Client tests #
+    #####################
+
+    def test_epiv2_crud(self):
```
Comment (shapiromatron) [inline tests/client/test_client.py:540]: i like this big test, very comprehensive

File: client/hawc_client/epiv2.py [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+        """
+        url = f"{self.session.root_url}/epidemiology/api/chemical/"
+        return self.session.post(url, data).json()
+
+    def update_chemical(self, chemical_id: int, data: dict) -> dict:
+        """
+        Update an existing chemical
+
+        Args:
+            chemical_id (int): chemical ID
```
Comment (shapiromatron) [inline client/hawc_client/epiv2.py:172]: remove extra newline between args

File: client/hawc_client/epiv2.py [hunk 7]
```diff
@@ -0,0 +1,538 @@
...
+        """
+        url = f"{self.session.root_url}/epidemiology/api/exposure/"
+        return self.session.post(url, data).json()
+
+    def update_exposure(self, exposure_id: int, data: dict) -> dict:
+        """
+        Update an existing exposure
+
+        Args:
+            exposure_id (int): exposure ID
```
Comment (shapiromatron) [inline client/hawc_client/epiv2.py:237]: remove newline between args; make other update methods consistent

File: client/hawc_client/epiv2.py [hunk 8]
```diff
@@ -0,0 +1,538 @@
...
+                significant (str): was the data extraction statistically significant? (predefined choices: 'NR', 'Yes', etc.)
+                group (str): results group for linking sets of results (free-text)
+                exposure_rank (int): value for ordering linked sets of results
+                exposure_transform (str): exposure transform (predefined choices: 'log10', 'squared', 'other', etc.)
+                outcome_transform (str): outcome transform (predefined choices: 'log10', 'squared', 'other', etc.)
+                confidence (str): overall confidence rating for the endpoint being extracted (free-text)
+                data_location (str): e.g. table number (free-text)
+                effect_description (str): description of the effect estimate with units, including comparison being made. (free-text)
+                statistical_method (str): brief description of the statistical analysis method (free-text)
+                comments (str): general comments (free-text)
```
Comment (shapiromatron) [inline client/hawc_client/epiv2.py:502]: add adjustment_factor to client; see https://github.com/shapiromatron/hawc/pull/902

**Review by shapiromatron — CHANGES_REQUESTED** [review]
looks close! A few minor updates, plus a new field added in https://github.com/shapiromatron/hawc/pull/902

File: client/hawc_client/epiv2.py (hunk 1 above)
Comment (DookTibs) [reply client/hawc_client/epiv2.py:12]: done

File: tests/client/test_client.py (hunk 2 above)
Comment (DookTibs) [reply tests/client/test_client.py:826]: sometimes I forget the nice things Python gives you :-). Done

File: tests/client/test_client.py (hunk 3 above)
Comment (DookTibs) [reply tests/client/test_client.py:803]: removed the extra explanation, i supposed variable names cover it. left the ones to stay consistent with the other create/read/etc. comments

File: tests/client/test_client.py (hunk 4 above)
Comment (DookTibs) [reply tests/client/test_client.py:817]: done

File: client/hawc_client/epiv2.py (hunk 7 above)
Comment (DookTibs) [reply client/hawc_client/epiv2.py:237]: done, I think I caught them all :)

File: client/hawc_client/epiv2.py (hunk 8 above)
Comment (DookTibs) [reply client/hawc_client/epiv2.py:502]: @shapiromatron you mean "adverse_direction" right? (assuming "adjustment_factor" was a typo)

Comment (shapiromatron) [reply client/hawc_client/epiv2.py:502]: yes! my bad, thank you for translating my error!

**Comment by DookTibs** [comment]
@shapiromatron getting CI errors around linting; when I run black locally on hawc_client/epiv2.py I don't get any. I have updated my repo so if there are modified configs in there I think I've got them; any advice on where else to look?

**Review by shapiromatron — APPROVED** [review]
LGTM! Nice work.

**Pull request merged**